.githooks/pre-push

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+# ensure that formatting is correct before pushing to the remote repository
+dotnet fake build -t CheckFormat
```

Comment:
Prepend with `dotnet tool restore` maybe?
I know very unlikely.

Reply:
Good call, I've added that. I also added documentation/specs for the 'api' of this git hook, as well as noted a potential performance optimization for the future.